lib/png/generate_png.js

```diff
@@ -1,12 +1,61 @@
 async (imgString) => {
+  const tempDiv = document.createElement("div")
+  tempDiv.innerHTML = imgString;
+
+  const asyncLoad = async (url) => {
+    const resp = await fetch(url);
+    if (resp.status !== 200) {
+      throw new Error("failed to fetch");
+    }
+    const blob = await resp.blob();
+    const f = new FileReader();
+    f.readAsDataURL(blob);
+    const promise = new Promise((resolve, reject) => {
+      f.addEventListener("load", () => {
+        resolve(f.result);
+      });
+      f.addEventListener("error", (e) => {
+        reject(e);
+      });
+    });
+    return promise;
+  };
+
+  const loadSVGImages = async () => {
+    const images = tempDiv.querySelectorAll("image");
+    const promises = [];
+    for (const image of images) {
+      const url = image.getAttribute("href");
```

Comment:
these may be local images. `file://`. it doesn't work right now but we'll want to. you should only fetch if it starts with `http`.